instance/tests/test_utils.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Utils module - Tests
+"""
+
+# Imports #####################################################################
+
+from tempfile import NamedTemporaryFile
+
+from instance.tests.base import TestCase
+from instance.utils import read_files
+
+
+# Tests #######################################################################
+
+class UtilsTestCase(TestCase):
+    """
+    Test cases for functions in the utils module
+    """
+    def test_read_files(self):
+        """
+        Ensure that the lines read are in the order they were written
+        """
+        with NamedTemporaryFile() as wfile1:
+            with NamedTemporaryFile() as wfile2:
+
+                rfile1 = open(wfile1.name, "rb")
+                rfile2 = open(wfile2.name, "rb")
```

Comment:
Ordinary files probably aren't the best candidates to test the `read_files()` function.  They are considered to be _always_ ready reading, so `poll()` won't do anything particularly useful for them.  Probably better to test with pipes, or even easier, a proper subprocess.  You could invoke a shell executing something like

    echo line1; echo line1 >&2; echo line2; echo line2 >&2; echo line3

to simulate a subprocess writing to both stdout and stderr.